src/Test.tsx

```diff
@@ -35,7 +37,9 @@ export default function Test() {
 							/>
 						</Grid.Column>
 						<Grid.Column width={2} >
-							<Button disabled={!isLoading} onClick={() => dispatchSourceTarget({ payload: {}, type: ActionsTypes.SWITCH_CHAINS })}><Icon fitted name='exchange' /></Button>
+							<div className='switchButton'>
+								<Button disabled={!isLoading} onClick={() => dispatchSourceTarget({ payload: {}, type: ActionsTypes.SWITCH_CHAINS })}><Icon fitted name='exchange' /></Button>
```

Comment:
Isn't using dynamic closures like `onClick` not recommended? The closure will be different with each render which forces the sub-tree to be re-rendered as well.

Reply:
is a fair point ... i have in my plans to include some memo as well , but on this commit they were not included. 
maybe this can be solved more easily by removing the dynamic closure